comp/core/flare/flareimpl/mock.go

```diff
@@ -0,0 +1,71 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build test
+
+// Package flareimpl implements the component flare
+package flareimpl
+
+import (
+	"net/http"
+
+	"github.com/DataDog/datadog-agent/comp/api/api"
+	"github.com/DataDog/datadog-agent/comp/core/flare"
+	"github.com/DataDog/datadog-agent/comp/core/flare/helpers"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+)
+
+// MockModule defines the fx options for the mock component.
+func MockModule() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(NewMock),
+	)
+}
+
+type provides struct {
+	fx.Out
+
+	Comp     flare.Component
+	Endpoint api.AgentEndpointProvider
+}
+
+// MockFlare is a mock of the
+type MockFlare struct{}
+
+type MockEndpoint struct {
+	Comp *MockFlare
+}
+
+func (MockEndpoint) Method() string {
```

Comment:
Are we using the endpoint provider in tests? Do we need to mock as part of the flare mock?

Reply:
yes I'm going to add a test with the api component for registering endpoints

Reply:
added in https://github.com/DataDog/datadog-agent/pull/24045/commits/71170e89b18fda9b8b70d9bc45164e364694839a